libbeat/publisher/pipeline/module.go

```diff
@@ -38,6 +38,12 @@ var publishDisabled = false
 
 const defaultQueueType = "mem"
 
+type Monitors struct {
+	Metrics   *monitoring.Registry
+	Telemetry *monitoring.Registry
```

Comment:
I prefer the proposed way instead of `map[string]*monitoring.Registry`, using a field reflect a clear intends, using a map complexify the access patterns instead of a nil check.

I would argue that maybe the monitoring package could expose a struct to list the possible monitors so we only have one place to change and the access definition are centralized.

Concerning having _Logger_ inside the monitor struct _Monitor_, at first I was against a logger is effectively some kind of monitor.

Reply:
I agree with PH, using a map will complicate the code and make it much more error prone. Instead we should have well defined types. Right now we use globals in a few places, so I kept the struct local to the publisher pipeline for now (makes me wonder if I missed some globals in the outputs).

Wasn't sure about the logger, but then decided to put it into the struct as well. Ideas was: all functionality passed/configured is for the purpose of getting some insights into an instance of the publisher pipeline. The metrics registry share the same type, but they are used in very different ways as well.